### PR TITLE
fix: add form relation options, list generation #719

### PR DESCRIPTION
**Pull request by SvyatovetsD**
affects: @haulmont/jmix-front-generator, @haulmont/jmix-react-core

### CODE REVIEW

File: packages/jmix-react-core/src/crud/getRelationOptions.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ export function getRelationOptions<
   Object.keys(data)
     .filter(queryName => {
       // Filter out query result related to the entity being listed so that only relation options are left
-      return (currentId || queryName !== getListQueryName(entityName))
+      return (loadFromEntityEditor || queryName !== getListQueryName(entityName))
```
Comment (vyacheslav-pushkin) [inline packages/jmix-react-core/src/crud/getRelationOptions.ts:19]: 1. I think this logic shouldn't be restricted to entity editor. In entity list we can filter by a relation field, so I think the same logic of collecting relation options should apply. 
2. In general, please avoid names like `loadFromEntityEditor`, arguments shall tell about intent rather than invocation context. E.g. `includeSelfReference` probably would be a better name.
3. The logic here is very complex and hard to follow. I would suggest to break it into separate booleans with clear names.

**Review by vyacheslav-pushkin — CHANGES_REQUESTED** [review]

File: packages/jmix-react-core/src/crud/getRelationOptions.ts (hunk 1 above)
Comment (SvyatovetsD) [reply packages/jmix-react-core/src/crud/getRelationOptions.ts]: 1. Too complex logic for low-priority use case scenario as discussed
2. Renamed to `includeSelfReference`
3. Replaced with named booleans
